viewer-prototype/src/browser/trace-viewer/components/utils/filtrer-tree/tree.tsx

```diff
@@ -59,7 +66,17 @@ export class FilterTree extends React.Component<FilterTreeProps, FilterTreeState
     };
 
     handleCollapse = (id: number): void => {
-        this.props.onToggleCollapse(id);
+        const nodes = sortNodes(this.state.filteredNodes, this.state.sortConfig);
+        this.props.onToggleCollapse(id, nodes);
```

Comment:
Why do you need the sorted nodes in the toggleCollapse, what is this parameter exactly? I'm not sure how to "javadoc" typescript stuff so users know the meaning of the parameter, but just document it somewhere and we'll make it official at some point.

Reply:
because to get visible ids in xy-chart-component to update orderedIds, I need the tree nodes and a list of collapsed nodes. Toggle collapse will change this list, so I send the nodes to parent component since it doesnt have it in its state. There may be a better way to do this, I'm open to suggestions.